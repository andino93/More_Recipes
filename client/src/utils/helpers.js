import jwtDecode from 'jwt-decode';
import history from '../utils/history';

const decodeToken = token => jwtDecode(token);

const logout = () => {
  window.sessionStorage.removeItem('token');
  history.push('/');
};

export {
  decodeToken,
  logout
};
