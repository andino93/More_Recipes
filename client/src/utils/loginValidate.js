import Validator from 'validator';
import isEmpty from 'lodash/isEmpty';

const loginValidate = (data) => {
  const errors = {};

  if (Validator.isEmpty(data.email)) {
    errors.email = 'Email is required';
  }

  if (Validator.isEmpty(data.password)) {
    errors.password = 'Password is required';
  }

  return {
    errors,
    isValid: isEmpty(errors)
  };
};

export default loginValidate;
