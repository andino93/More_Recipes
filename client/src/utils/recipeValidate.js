import Validator from 'validator';
import isEmpty from 'lodash/isEmpty';

const recipeValidate = (data) => {
  const name = data.name.trim(),
    description = data.description.trim(),
    ingredient = data.ingredient.trim();
  const errors = {};

  if (!Validator.isEmpty(name)) {
    const hasNumber = name.split('').filter(character => Validator.toInt(character));
    if (hasNumber.length !== 0) {
      errors.name = 'Recipe name must contain alphabetic characters only';
    }
  } else { errors.name = 'Recipe name is required'; }

  if (!Validator.isEmpty(description)) {
    if (!Validator.isLength(description, { min: 4, max: undefined })) {
      errors.description = 'Recipe description must be at least 4 characters';
    }
  } else { errors.description = 'Recipe description is required'; }

  if (!Validator.isEmpty(ingredient)) {
    if (!Validator.isLength(ingredient, { min: 3, max: undefined })) {
      errors.ingredient = 'Recipe ingredient must be at least 3 characters';
    }
  } else { errors.ingredient = 'Recipe ingredient is required'; }

  return {
    errors,
    isValid: isEmpty(errors)
  };
};

export default recipeValidate;