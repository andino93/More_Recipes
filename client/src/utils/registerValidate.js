import Validator from 'validator';
import isEmpty from 'lodash/isEmpty';

const registerValidate = (data) => {
  const errors = {};
  
  if (Validator.isEmpty(data.firstName)) {
    errors.firstName = 'First name is required';
  }

  if (Validator.isEmpty(data.lastName)) {
    errors.lastName = 'Last name is required';
  }

  if (Validator.isEmpty(data.username)) {
    errors.username = 'Username is required';
  }

  if (Validator.isEmpty(data.email)) {
    errors.email = 'Email is required';
  }

  if (Validator.isEmpty(data.password)) {
    errors.password = 'Password is required';
  }

  if (Validator.isEmpty(data.confirmPassword)) {
    errors.confirmPassword = 'Confirm password is required';
  }

  if (!Validator.equals(data.password, data.confirmPassword)) {
    errors.confirmPassword = 'Passwords must match';
  }

  return {
    errors,
    isValid: isEmpty(errors)
  };
};

export default registerValidate;
