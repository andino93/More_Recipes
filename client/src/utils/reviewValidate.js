import Validator from 'validator';
import isEmpty from 'lodash/isEmpty';

const reviewValidate = (data) => {
  const content = data.content.trim();
  const errors = {};

  if (!Validator.isEmpty(content)) {
    if (!Validator.isLength(content, { min: 4, max: undefined })) {
      errors.content = 'Review  must be at least 4 characters';
    }
  } else { errors.content = 'Review details is required'; }

  return {
    errors,
    isValid: isEmpty(errors)
  };
};

export default reviewValidate;