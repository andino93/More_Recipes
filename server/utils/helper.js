const errorHandler = (error, res) => {
  switch (error.code) {
    case 404:
      return res.status(404).json({ message: error.message });
    case 500:
      return res.status(400).json({ message: error.message });
    default:
      return res.status(500).json(error);
  }
};

const successHandler = (code, body, res) => {
  switch (code) {
    case 201:
      return res.status(201).json({ body });
    default:
      return res.status(200).json({ body });
  }
};

const generatePaginationMeta = (dbResult, limit, offset) => {
  const paginationMeta = {
    pageCount: Math.ceil(dbResult.count / limit),
    totalCount: dbResult.count,
    outputCount: dbResult.rows.length,
    pageSize: limit,
    currentPage: Math.floor(offset / limit) + 1
  };
  return paginationMeta;
};

export { errorHandler, successHandler, generatePaginationMeta };
