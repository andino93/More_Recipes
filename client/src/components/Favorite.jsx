import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import PropTypes from 'react-proptypes';
import ReactPaginate from 'react-paginate';
import { getAllFavorites } from '../actions/recipe/recipeActions';
import Header from './partials/Headers/Header.jsx';
import FooterComponent from './partials/Footer.jsx';
import FavoriteRecipe from './recipe/FavoriteRecipe.jsx';
import { recipeResponseType } from './../utils/helpers';


/**
 *
 * @class Favorite
 *
 * @extends {React.Component}
 */
class Favorite extends Component {
  /**
   *
   * @param {props} props
   */
  constructor(props) {
    super(props);
    this.state = {
      favorites: [],
      pageCount: 0,
      page: 1
    };
    this.handlePageChange = this.handlePageChange.bind(this);
  }
  /**
   * @returns {XML} XML/JSX
   *
   * @memberof Favorite
   */
  componentWillMount() {
    this.props.getAllFavorites(this.state.page);
  }
  /**
   *
   * @param  {object} nextProps
   *
   * @returns {XML} XML/JSX
   *
   * @memberof Favorite
   */
  componentWillReceiveProps(nextProps) {
    const { responseType, favorites, pagination } = nextProps.favoriteState;
    if (responseType === recipeResponseType.GET_ALL_FAVORITES_SUCCESS) {
      this.setState({
        favorites,
        pageCount: pagination.pageCount || 0
      });
    }
  }
  /**
   * Handles page change
   *
   * @method handlePageChange
   *
   * @param {event} page
   *
   * @return {void}
   */
  handlePageChange(page) {
    const currentPage = page.selected + 1;
    this.props.getAllFavorites(currentPage);
  }
  /**
   *
   * @returns {XML} XML/JSX
   *
   * @memberof Favorite
   */
  render() {
    const { favorites } = this.state;
    return (
      <div>
        <Header/>
        <div className="wrapper">
          <div className="container">
            <nav aria-label="breadcrumb" className="mt-40 mb-10" >
              <ol className="breadcrumb bg-white">
                <li className="breadcrumb-item"><Link to="/">Home</Link></li>
                <li className="breadcrumb-item active" aria-current="page">
                Most Favourited</li>
              </ol>
            </nav>
          </div>
          <div className="container mb-20 recipe-details-container">
            <div className="row">
              {
                favorites.map((favorite, index) => (
                  <FavoriteRecipe
                    key={index}
                    recipe={favorite.Recipe}
                  />
                ))
              }
            </div>
          </div>
          <div className="clearfix" />
          <ReactPaginate
            pageCount={this.state.pageCount}
            pageRangeDisplayed={2}
            marginPagesDisplayed={2}
            onPageChange={this.handlePageChange}
            containerClassName="pagination justify-content-center"
            subContainerClassName="page-item"
            pageLinkClassName="page-link"
            activeClassName="page-item active"
            nextLinkClassName="page-link"
            previousLinkClassName="page-link"
          />
        </div>
        <FooterComponent/>
      </div>
    );
  }
}

Favorite.propTypes = {
  getAllFavorites: PropTypes.func.isRequired,
  favoriteState: PropTypes.object
};

const mapStateToProps = state => ({
  favoriteState: state.recipeReducer
});

const mapDispatchToProps = dispatch =>
  bindActionCreators({ getAllFavorites }, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(Favorite);