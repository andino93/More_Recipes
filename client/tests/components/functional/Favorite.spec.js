import React from 'react';
import expect from 'expect';
import Adapter from 'enzyme-adapter-react-16';
import { configure } from 'enzyme';
import mockStore from '../../__mocks__/mockStore';
import mountWrap from '../../__mocks__/utils';
import Favorite from '../../../src/components/Favorite.jsx';
import localStorageMock from '../../__mocks__/localStorageMock';
import mockData from "../../__mocks__/mockData";
import { recipeResponseType } from '../../../src/utils/helpers';

configure({ adapter: new Adapter() });

// noinspection JSAnnotator
window.localStorage = localStorageMock;
window.localStorage.token = mockData.token;

describe('<Favorite>', () => {
  const store = mockStore({});
  const connectedFavorite = mountWrap(Favorite, store, {});

  it('should render Favorite component', (done) => {
    expect(connectedFavorite).toMatchSnapshot();
    done();
  });

  it('should check for componentWillReceiveProps', () => {
    connectedFavorite.find('Favorite').instance().componentWillReceiveProps({
      favoriteState: {
        responseType: recipeResponseType.GET_ALL_FAVORITES_SUCCESS,
        favorites: [],
        pagination: 1
      },
    });
  });
});