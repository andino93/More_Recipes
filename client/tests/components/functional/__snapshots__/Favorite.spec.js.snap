// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Favorite> should render Favorite component 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "blank",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Connect(Favorite)>
        <Favorite
          getAllFavorites={[Function]}
        >
          <div>
            <Connect(Header)>
              <Header
                logoutAction={[Function]}
              >
                <div>
                  <nav
                    className="navbar navbar-expand-sm navbar-light bg-navbar fixed-top"
                  >
                    <Link
                      className="navbar-brand"
                      replace={false}
                      to="/"
                    >
                      <a
                        className="navbar-brand"
                        href="/"
                        onClick={[Function]}
                      >
                        More Recipes
                      </a>
                    </Link>
                    <button
                      aria-controls="navbarSupportedContent"
                      aria-expanded="false"
                      aria-label="Toggle navigation"
                      className="navbar-toggler"
                      data-target="#navbarSupportedContent"
                      data-toggle="collapse"
                      type="button"
                    >
                      <span
                        className="navbar-toggler-icon"
                      />
                    </button>
                    <div
                      className="collapse navbar-collapse"
                      id="navbarSupportedContent"
                    >
                      <ul
                        className="navbar-nav ml-auto"
                      >
                        <li
                          className="nav-item dropdown"
                        >
                          <Link
                            aria-expanded="false"
                            aria-haspopup="true"
                            className="nav-link dropdown-toggle"
                            data-toggle="dropdown"
                            id="navbarDropdown"
                            replace={false}
                            role="button"
                            to="#"
                          >
                            <a
                              aria-expanded="false"
                              aria-haspopup="true"
                              className="nav-link dropdown-toggle"
                              data-toggle="dropdown"
                              href="#"
                              id="navbarDropdown"
                              onClick={[Function]}
                              role="button"
                            >
                              My Account
                            </a>
                          </Link>
                          <div
                            aria-labelledby="navbarDropdown"
                            className="dropdown-menu dropdown-menu-right mr-20"
                          >
                            <Link
                              className="dropdown-item"
                              replace={false}
                              to="/profile"
                            >
                              <a
                                className="dropdown-item"
                                href="/profile"
                                onClick={[Function]}
                              >
                                <i
                                  className="fa fa-user-circle"
                                />
                                 My Profile
                              </a>
                            </Link>
                            <Link
                              className="dropdown-item"
                              replace={false}
                              to="/user-recipes"
                            >
                              <a
                                className="dropdown-item"
                                href="/user-recipes"
                                onClick={[Function]}
                              >
                                <i
                                  className="fa fa-list"
                                />
                                 My Recipes
                              </a>
                            </Link>
                            <Link
                              className="dropdown-item"
                              replace={false}
                              to="/my-favorite"
                            >
                              <a
                                className="dropdown-item"
                                href="/my-favorite"
                                onClick={[Function]}
                              >
                                <i
                                  className="fa fa-star"
                                />
                                 My Favorites
                              </a>
                            </Link>
                            <Link
                              className="dropdown-item"
                              replace={false}
                              to="/addRecipe"
                            >
                              <a
                                className="dropdown-item"
                                href="/addRecipe"
                                onClick={[Function]}
                              >
                                <i
                                  className="fa fa-plus"
                                />
                                 Add Recipes
                              </a>
                            </Link>
                            <div
                              className="dropdown-divider"
                            />
                            <Link
                              className="dropdown-item"
                              onClick={[Function]}
                              replace={false}
                              to="#"
                            >
                              <a
                                className="dropdown-item"
                                href="#"
                                onClick={[Function]}
                              >
                                <i
                                  className="fa fa-sign-out"
                                />
                                Log out
                              </a>
                            </Link>
                          </div>
                        </li>
                      </ul>
                    </div>
                  </nav>
                </div>
              </Header>
            </Connect(Header)>
            <div
              className="wrapper"
            >
              <div
                className="container"
              >
                <nav
                  aria-label="breadcrumb"
                  className="mt-40 mb-10"
                >
                  <ol
                    className="breadcrumb bg-white"
                  >
                    <li
                      className="breadcrumb-item"
                    >
                      <Link
                        replace={false}
                        to="/"
                      >
                        <a
                          href="/"
                          onClick={[Function]}
                        >
                          Home
                        </a>
                      </Link>
                    </li>
                    <li
                      aria-current="page"
                      className="breadcrumb-item active"
                    >
                      Most Favourited
                    </li>
                  </ol>
                </nav>
              </div>
              <div
                className="container mb-20 recipe-details-container"
              >
                <div
                  className="row"
                />
              </div>
              <div
                className="clearfix"
              />
              <PaginationBoxView
                activeClassName="page-item active"
                breakLabel="..."
                containerClassName="pagination justify-content-center"
                disableInitialCallback={false}
                disabledClassName="disabled"
                marginPagesDisplayed={2}
                nextClassName="next"
                nextLabel="Next"
                nextLinkClassName="page-link"
                onPageChange={[Function]}
                pageCount={0}
                pageLinkClassName="page-link"
                pageRangeDisplayed={2}
                previousClassName="previous"
                previousLabel="Previous"
                previousLinkClassName="page-link"
                subContainerClassName="page-item"
              >
                <ul
                  className="pagination justify-content-center"
                >
                  <li
                    className="previous disabled"
                  >
                    <a
                      className="page-link"
                      onClick={[Function]}
                      onKeyPress={[Function]}
                      tabIndex="0"
                    >
                      Previous
                    </a>
                  </li>
                  <li
                    className="next"
                  >
                    <a
                      className="page-link"
                      onClick={[Function]}
                      onKeyPress={[Function]}
                      tabIndex="0"
                    >
                      Next
                    </a>
                  </li>
                </ul>
              </PaginationBoxView>
            </div>
            <FooterComponent>
              <footer
                className="p-10 well well-sm m-0 shadow-lite"
              >
                <div
                  className="text-center text-muted"
                >
                  <p
                    className="bold"
                  >
                    © 2017 More Recipes
                  </p>
                </div>
              </footer>
            </FooterComponent>
          </div>
        </Favorite>
      </Connect(Favorite)>
    </Router>
  </BrowserRouter>
</Provider>
`;
